framework/src/transfers/MultiAppReporterTransfer.C

```diff
@@ -133,15 +148,58 @@ MultiAppReporterTransfer::executeFromMultiapp()
     return;
 
-  unsigned int ind = _subapp_index != std::numeric_limits<unsigned int>::max() ? _subapp_index : 0;
-  if (getFromMultiApp()->hasLocalApp(ind) &&
-      (!hasToMultiApp() || getToMultiApp()->hasLocalApp(ind)))
-    for (unsigned int n = 0; n < _from_reporter_names.size(); ++n)
-      transferReporter(_from_reporter_names[n],
-                       _to_reporter_names[n],
-                       getFromMultiApp()->appProblemBase(ind),
+  std::vector<unsigned int> indices;
+  if (_subapp_index == std::numeric_limits<unsigned int>::max() || _distribute_reporter_vector)
+  {
+    indices.resize(getFromMultiApp()->numGlobalApps());
+    std::iota(indices.begin(), indices.end(), 0);
+  }
+  else
+    indices = {_subapp_index};
+
+  if (_distribute_reporter_vector)
+    for (const auto & ind : indices)
+      for (unsigned int n = 0; n < _to_reporter_names.size(); ++n)
+      {
+        auto size = getFromMultiApp()->numGlobalApps();
+        clearVectorReporter(_to_reporter_names[n],
+                            hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
+                                            : getFromMultiApp()->problemBase());
+        resizeReporter(_to_reporter_names[n],
                        hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
-                                       : getFromMultiApp()->problemBase());
+                                       : getFromMultiApp()->problemBase(),
+                       size);
+      }
+
+  for (const auto & ind : indices)
+    if (getFromMultiApp()->hasLocalApp(ind) &&
+        (!hasToMultiApp() || getToMultiApp()->hasLocalApp(ind)))
+      for (unsigned int n = 0; n < _from_reporter_names.size(); ++n)
+      {
+        if (_distribute_reporter_vector)
+        {
+          if (getFromMultiApp()->appProblemBase(ind).processor_id() == 0) // Subapp Root Rank only
+            transferToVectorReporter(_from_reporter_names[n],
+                                     _to_reporter_names[n],
+                                     getFromMultiApp()->appProblemBase(ind),
+                                     hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
+                                                     : getFromMultiApp()->problemBase(),
+                                     ind);
+        }
+        else
+          transferReporter(_from_reporter_names[n],
+                           _to_reporter_names[n],
+                           getFromMultiApp()->appProblemBase(ind),
+                           hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
+                                           : getFromMultiApp()->problemBase());
+      }
+  if (_distribute_reporter_vector)
+    for (unsigned int n = 0; n < _to_reporter_names.size(); ++n)
+    {
+      sumVectorReporter(_to_reporter_names[n],
+                        hasToMultiApp() ? getToMultiApp()->appProblemBase(0)
+                                        : getFromMultiApp()->problemBase());
+    }
```

Comment:
I need to call sumVectorReporter for every reporter with every processor
one time. I am having some trouble understanding if
`getToMultiApp()->appProblemBase(i)` will be the same problemBase for every i.
If so then me setting i to 0 will be fine if not I need to rethink the logic
here. What I am trying to prevent is that proc 0 has 2 local apps and proc 1 has
1 local app, so then they will call the function different amount of times if I
looped over each indices.

Reply:
Will it hang or just give the wrong answer if it is called twice on the same processor?

Reply:
It will hang.

Reply:
@GiudGiud I think I need to disable sibling transfers, when using the distributed vector functionality. Would you still want this MultiappReporterTransfer? No change if not using the distributed functionality.

Reply:
I have the errors setup for distributed reporters now!